[Darren Mok] iP

deetoomoo opened the pull request:
### BobbyBot

> With BobbyBot, you'll never miss a deadline ever again!


**Features**
- Keep track of tasks and deadlines
- Persistence data using .txt file

**Steps to get started**
1. Download jar file from release from [here ](www.google.com)
2. Double click jar file to start
3. Start adding your tasks! 😋

**Features**

- [x] Manage tasks
- [ ] Manage deadlines with specific datetime
- [ ] Find all deadlines on a datetime


Main method (for Java programmers)
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

deetoomoo committed:
.gitignore: Add line to ignore all .class files
Duke: Add echo feature and program introduction

deetoomoo committed:
Task: New class for task objects
Duke: Add ability to store tasks and print back to user when requested

deetoomoo committed:
Task: Add ability to mark tasks as done

deetoomoo committed:
Task: Use Polymorphism to represent Event, Deadline and ToDo objects

deetoomoo committed:
Duke: Correct print statement in main for Event and Deadline

deetoomoo committed:
text-ui-test: Create first test case

deetoomoo committed:
Duke: Ensure that bye statement terminates program

deetoomoo committed:
Duke: Remove whitespaces

deetoomoo committed:
Duke: Change nested if-else statement in main to switch

deetoomoo committed:
Abstract Main into BobbyBot to handle user inputs

deetoomoo committed:
Exceptions: Add exceptions for exception handling

deetoomoo committed:
BobbyBot: Add javadocs comments

deetoomoo committed:
Main: Add line div for exception handling reply to user

deetoomoo committed:
Text-ui-test: Change test to call Main instead of Duke (rename)

deetoomoo committed:
BobbyBot: Remove task type from description of tasks

deetoomoo committed:
Exceptions: Add more exception handling for user input

deetoomoo committed:
BobbyBot: Use ArrayList to store tasks instead of primitive arrays

deetoomoo committed:
Bobbybot: Add delete feature

deetoomoo committed:
Test: Update test cases to test deletion

deetoomoo committed:
Refactor: Use enum for bot commands

deetoomoo committed:
Tests: fix typo

deetoomoo committed:
Tests: fix typo again

deetoomoo committed:
Task: remove unnecessary this. to follow style guide

deetoomoo committed:
Feat: Implement loading of saved tasks

deetoomoo committed:
Feat: Save after each action

deetoomoo committed:
Feat: Add datetime for Deadlines

deetoomoo committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Feat: Add datetime for Deadlines

# Conflicts:
#	src/main/java/BobbyBot.java: resolved by keeping both

deetoomoo committed:
Save: Allow datetime formats to be saved in database.txt

deetoomoo committed:
Refactor: Extract Storage class to handle database

deetoomoo committed:
Refactor: Extract TaskList class from BobbyBot

deetoomoo committed:
Refactor: Extract Ui class

deetoomoo committed:
Refactor: Continue extracting TaskList; deletion and creation

deetoomoo committed:
Refactor: Extract save function from BobbyBot to Storage

deetoomoo committed:
Ui: Cleaning up ui class

deetoomoo committed:
Refactor: Add create event to TaskList

deetoomoo committed:
Refactor: Extract Parser class

deetoomoo committed:
Refactor: Clean up code and rename DukeExceptions to BobbyBotExceptions

deetoomoo committed:
Refactor: Remove unnecessary ui.showLine()

deetoomoo committed:
Refactor: Move all classes into bobbybot package

deetoomoo committed:
Test: Add Junit tests to package

deetoomoo committed:
JAR: Generate JAR executable file

deetoomoo committed:
Javadocs: Add Java docs to TaskList

deetoomoo committed:
Style: Clean up code to follow standard

deetoomoo committed:
Feat: Add feature to find for Level-9

deetoomoo committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Javadocs: Add Java docs to TaskList

deetoomoo committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Feat: Add feature to find for Level-9

deetoomoo committed:
JavaDoc: Added Javadocs for remaining classes

deetoomoo committed:
Test: Add more tests for public classes

deetoomoo committed:
Refactor: Add javadocs and change access of a helper class to private

deetoomoo committed:
Feat: Improve find feature

deetoomoo committed:
Test: Add test functions for find feature

syoopie commented on src/java/bobbybot/BobbyBot.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Constructor to initialise classes
+     */
+    public BobbyBot()  {
```
There seems to be an extra white space here?

deetoomoo replied:
Yep! Thanks for the comment. Will fix it!

syoopie commented on src/java/bobbybot/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+/** Represents a bobbybot. Task that should be completed by a specified Date and Time
+ *  @author mokdarren
+ */
+public class Deadline extends Task{
```
There seems to be a missing whitespace after Task. There are multiple occurrences of this throughout, so I'll leave it to you to find them :D

syoopie commented on src/java/bobbybot/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+     * @param by date and time that the task should be completed by
+     * @param isDone boolean flag if task is done
+     */
+    public Deadline(String description, String by, boolean isDone, DateTimeFormatter formatter) {
```
Perhaps the name `by` could be something more descriptive with respect to the context? Maybe something like `doneBy`

syoopie commented on src/java/bobbybot/Event.java:
```diff
@@ -0,0 +1,43 @@
+     * @param description description of task
+     * @param at time period of bobbybot.Event (start-end)
+     */
+    public Event(String description, String at)  {
```
Same comment as the usage of `by`. Perhaps using `eventTiming` would be better than `at`.

syoopie commented on src/java/bobbybot/ToDo.java:
```diff
@@ -0,0 +1,30 @@
+/** Represents a task without any time restrictions
+ *  @author mokdarren
+ */
+public class ToDo extends Task{
```
Since "todo" is one word, perhaps naming it `Todo` would be more appropriate than `ToDo`.

syoopie reviewed (commented):
Good adherence to coding standard overall! Just a few tiny errors.

peilinye commented on src/java/bobbybot/Storage.java:
```diff
@@ -0,0 +1,70 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class Storage {
```
Perhaps a header comment here for this class could help to explain the functionality of this class?

peilinye commented on src/java/bobbybot/Parser.java:
```diff
@@ -0,0 +1,93 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class Parser {
```
Perhaps a header comment here for this class could help to explain the functionality of this class?

peilinye commented on src/java/bobbybot/Task.java:
```diff
@@ -0,0 +1,46 @@
+package bobbybot;
+
+/** Represents a task that can be completed
```
Should the comment be in the next line after `/**`? Same issue found in `ToDo` class

peilinye commented on src/java/bobbybot/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+        }
+    }
+
+    public void find(String keyword) {
```
Perhaps you could include a header comment for this method, and have a clearer method name such as `findMatchingTasks`?

peilinye commented on src/java/bobbybot/ToDo.java:
```diff
@@ -0,0 +1,30 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
Perhaps including a new line here would be good?

peilinye reviewed (commented):
Overall the naming of classes, methods, and variables follows the guidelines given. Only some small issues were found. :)

deetoomoo committed:
Style: Fix java coding standard comments from pr

deetoomoo committed:
Style: Add function headers and make functions more descriptive

deetoomoo committed:
Style: Add class header docs for Exceptions

deetoomoo committed:
Merge branch 'add-gradle-support' of https://github.com/mokdarren/ip into branch-A-Gradle

deetoomoo committed:
Refactor: Move files to suit gradle file structure

deetoomoo committed:
Gradle: Add support for Checkstyle

deetoomoo committed:
Style: Make changes according to checkstyle.xml

deetoomoo committed:
Merge branch 'branch-A-Gradle'

Added Gradle support for Intelli-J IDEA and fix checkstyle errors

deetoomoo committed:
Style: Fix Checkstyle errors for test files

deetoomoo committed:
Refactor: Change test database file to suit new structure after Gradle

deetoomoo committed:
Storage: Fix bug where program dies if no data or .txt data file

deetoomoo committed:
Test: Fix storage tests to suit gradle Junit test

deetoomoo committed:
Refactor: Group classes into packages

deetoomoo committed:
Feat: Implement barebones GUI interface with echo

deetoomoo committed:
Refactor: extract adding command logic to AddCommand class

deetoomoo committed:
Refactor: Extract bye command log to ExitCommand

deetoomoo committed:
Refactor: Extract list logic to ListCommand

deetoomoo committed:
Refactor: Extract Done logic to DoneCommand class

deetoomoo committed:
Refactor: Extract Find logic to FindCommand

deetoomoo committed:
Refactor: Extract Delete and Done logic to Command classes

deetoomoo committed:
Feat: GUI is integrated with cli

deetoomoo committed:
Style: Change font size for GUI

deetoomoo committed:
Merge branch 'branch-Level-10'
Completed GUI integration

deetoomoo committed:
Jar: Made some changes to run Shadow Jar
Jar file path: /build/libs/bobbybot.jar

deetoomoo committed:
Bug: Fix bug where text is truncated in Dialogbox

deetoomoo committed:
Style: Fixed all checkstyle related to imports

deetoomoo committed:
Style: Add reply to user about task deleted

deetoomoo committed:
Style: Make font size smaller

deetoomoo committed:
Refactor: Separate logic of execution from getResponse

deetoomoo committed:
Javadocs: Remove extra @return tags for void function

deetoomoo committed:
Test: Fix TaskList tests and Storage test

deetoomoo committed:
Bug Fix: Typo

deetoomoo committed:
Feature: Add Assertions to bobbybot

deetoomoo committed:
Merge pull request #1 from mokdarren/branch-A-Assertions

Branch a assertions

deetoomoo committed:
Refactor: Improve code quality

deetoomoo committed:
Merge pull request #2 from mokdarren/branch-A-CodeQuality

Refactor: Improve code quality

deetoomoo committed:
Test: Add more test cases for Parser and fixed some bugs

deetoomoo committed:
Test: Add tests for AddCommand

deetoomoo committed:
Test: Add test for DoneCommand

deetoomoo committed:
DoneCommandTest: Add cleanup function

deetoomoo committed:
Test: Add delete command tests

deetoomoo committed:
Test: Add tests for List and Find

deetoomoo committed:
Test: Add more assertions

deetoomoo committed:
Feature: Support adding of persons with details to contact list
Only supports tasks, deadlines and events as objects users can interact with.

However users might need to keep track of their contacts as well

Let’s bring in a new feature to keep track of contacts

deetoomoo committed:
Add Ui.png

deetoomoo committed:
Set theme jekyll-theme-cayman

deetoomoo committed:
Parser: Added more tests for contact command and regex checks

deetoomoo committed:
Merge branch 'master' of https://github.com/mokdarren/ip

* 'master' of https://github.com/mokdarren/ip:
  Set theme jekyll-theme-cayman

deetoomoo committed:
Style: Fixed remaining checkstyle problems

deetoomoo committed:
Refactor: Move command tests into a package

deetoomoo committed:
Update README.md

deetoomoo committed:
Update README.md

deetoomoo committed:
DeleteContactCommand: Add feature to delete contacts

deetoomoo committed:
Merge branch 'master' of https://github.com/mokdarren/ip

deetoomoo committed:
Feature: Support saving and loading of contacts

deetoomoo committed:
Update README.md

deetoomoo committed:
Update README.md

deetoomoo committed:
Update README.md

deetoomoo committed:
Javadocs: Fix header comments

deetoomoo committed:
Update sample data

deetoomoo committed:
Update README.md

deetoomoo committed:
Update README.md

Fix table

deetoomoo committed:
Update README.md

Fix table
